gpsdio_segment/core.py

```diff
@@ -1,103 +1,106 @@
 
-import pyproj
 
 
-from gpsdio_segment.segment import Segment, BadSegment
-from gpsdio_segment.segment import Segment, NoiseSegment
-from gpsdio_segment.state import SegmentState
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.WARNING)
 
+log = logger.info
 
-logger = logging.getLogger(__file__)
-# logger.setLevel(logging.DEBUG)
+inf = float("inf")
 
+POSITION_TYPES = {
+    'AIS.1' : {'AIS-A'}, 
+    'AIS.2' : {'AIS-A'},
+    'AIS.3' : {'AIS-A'},
+    'AIS.18' : {'AIS-B'}, 
+    'AIS.19' : {'AIS-B'},
+    'AIS.27' : {'AIS-A', 'AIS-B'}
+    } 
 
-# See Segmentizer() for more info
-DEFAULT_MAX_HOURS = 24  # hours
-DEFAULT_MAX_SPEED = 25  # knots
-DEFAULT_NOISE_DIST = 100  # nautical miles
-INFINITE_SPEED = 1000000
-REPORTED_SPEED_MULTIPLIER = 1.1 # multiply this by reported speed in max speed calculation
-MAX_SPEED_MULTIPLIER = 1600     # magic number used to compute max_speed_at_distance
-MAX_SPEED_EXPONENT = 2.0     # magic number used to compute max_speed_at_distance
+INFO_TYPES = {
+    'AIS.5' : 'AIS-A',
+    'AIS.19' : 'AIS-B', 
+    'AIS.24' : 'AIS-B'
+    }
+
+INFO_PING_INTERVAL_MINS = 6
 
 # The values 52 and 102.3 are both almost always noise, and don't
 # reflect the vessel's actual speed. They need to be commented out.
 # The value 102.3 is reserved for "bad value." It looks like 51.2
-# is also almost always noise. Because the values are floats,
+# is also almost always noise. The value 63 means unavailable for
+# type 27 messages so we exclude that as well. Because the values are floats,
 # and not always exactly 102.3 or 51.2, we give a range.
-REPORTED_SPEED_EXCLUSION_RANGES = [(51.0, 51.3),(102.2,103.0)]
-AIS_CLASS = {
-    1: 'A',
-    2: 'A',
-    3: 'A',
-    5: 'A',
-    18: 'B',
-    19: 'B',
-    24: 'B'
-}
-
-class Segmentizer(object):
+REPORTED_SPEED_EXCLUSION_RANGES = [(51.15, 51.25), (62.95, 63.05), (102.25,102.35)]
+SAFE_SPEED = min([x for (x, y) in REPORTED_SPEED_EXCLUSION_RANGES])
+
+
+POSITION_MESSAGE = object()
+INFO_MESSAGE = object()
+BAD_MESSAGE = object()
+
+NO_MATCH = object()
+IS_NOISE = object()
+
+
+class Segmentizer(DiscrepancyCalculator):
 
     """
     Group positional messages into related segments based on speed and distance.
     """
 
-    def __init__(self, instream, mmsi=None, max_hours=DEFAULT_MAX_HOURS,
-                 max_speed=DEFAULT_MAX_SPEED, noise_dist=DEFAULT_NOISE_DIST,
-                 reported_speed_multiplier=REPORTED_SPEED_MULTIPLIER,
-                 max_speed_multiplier = MAX_SPEED_MULTIPLIER,
-                 max_speed_exponent=MAX_SPEED_EXPONENT,
-                 collect_match_stats=False
+    max_hours = 24
+    penalty_hours = 6
+    hours_exp = 0.5
+    buffer_hours = 0.25
+    lookback = 5
+    lookback_factor = 2
+    max_knots = 25
+    ambiguity_factor = 10.0
+    short_seg_threshold = 10
+    transponder_mismatch_weight = 0.1
+    penalty_speed = 5.0
+    max_open_segments = 20
+    min_type_27_hours = 1.0
```

Comment:
It good be nice to know how you calculated the default values here.

Reply:
I added a note about how we looked at a large number of vessel tracks to come up with these values.